components/match2/wikis/brawlhalla/match_group_input_custom.lua

```diff
@@ -9,405 +9,172 @@
+			return CustomMatchGroupInput.getParticipantsOfOpponent(map, opponent, opponentIndex)
+		end)
+		-- Match/Subobjects:luaGetMap sets a empty table as default value for participants.
+		-- Once subobjects have been refactored away this can be removed.
+		map.participants = nil
+
+		local opponentInfo = Array.map(matchOpponents, function(_, opponentIndex)
+			local score, status = MatchGroupInputUtil.computeOpponentScore({
+				walkover = map.walkover,
+				winner = map.winner,
+				opponentIndex = opponentIndex,
+				score = map['score' .. opponentIndex],
+			}, CustomMatchGroupInput.calculateMapScore(map.winner, map.finished))
+			return {score = score, status = status}
+		end)
+
+		map.scores = Array.map(opponentInfo, Operator.property('score'))
+		if map.finished then
+			map.resulttype = MatchGroupInputUtil.getResultType(winnerInput, finishedInput, opponentInfo)
+			map.walkover = MatchGroupInputUtil.getWalkover(map.resulttype, opponentInfo)
+			map.winner = MatchGroupInputUtil.getWinner(map.resulttype, winnerInput, opponentInfo)
 		end
-		map.walkover = Table.includes(ALLOWED_STATUSES, map.walkover) and map.walkover or 'L'
-		map.scores = {-1, -1}
-		map.resulttype = 'default'
 
-		return map
+		table.insert(maps, map)
+		match[key] = nil
 	end
 
-	if hasManualScores then
-		for scoreIndex, _ in Table.iter.spairs(indexedScores, CustomMatchGroupInput._placementSortFunction) do
-			if not tonumber(map.winner) then
-				map.winner = scoreIndex
-			else
-				break
-			end
-		end
-
-		return map
-	end
+	return maps
+end
 
-	if map.winner == 'skip' then
-		map.scores = {-1, -1}
-		map.resulttype = 'np'
-	elseif winnerInput == 1 then
-		map.scores = {1, 0}
-	elseif winnerInput == 2 then
-		map.scores = {0, 1}
-	elseif winnerInput == 0 or map.winner == 'draw' then
-		map.scores = {0, 0}
-		map.resulttype = 'draw'
+---@param maps table[]
+---@return fun(opponentIndex: integer): integer
+function CustomMatchGroupInput.calculateMatchScore(maps)
+	return function(opponentIndex)
+		return MatchGroupInputUtil.computeMatchScoreFromMapWinners(maps, opponentIndex)
 	end
-
-	return map
 end
 
 ---@param map table
----@param match table
----@param numberOfOpponents integer
-function CustomMatchGroupInput.processPlayerMapData(map, match, numberOfOpponents)
-	local participants = {}
-	for opponentIndex = 1, numberOfOpponents do
-		local opponent = match['opponent' .. opponentIndex]
-		if opponent.type == Opponent.solo then
-			CustomMatchGroupInput._processSoloMapData(opponent.match2players[1], map, opponentIndex, participants)
-		end
+---@param opponent table
+---@param opponentIndex integer
+---@return table<string, table>?
+function CustomMatchGroupInput.getParticipantsOfOpponent(map, opponent, opponentIndex)
+	if opponent.type == Opponent.solo then
+		CustomMatchGroupInput._processSoloMapData(opponent.match2players[1], map, opponentIndex)
 	end
-
-	map.participants = participants
+	return nil
```

Comment:
intended to not support duos, trios, quads nor teams?

Reply:
Intended to onlu have participants on solos atm, yet